### ISSUE TITLE
Record as return type - no parameterless constructor defined

### ISSUE THREAD

**Issue by spaasis** [jbogard/MediatR#713]
Hi and thanks for the brilliant library!

I recently started to convert our DTOs to records, but MediatR can't apparently handle parameterless constructors in return types:

Paraphrasing:
`public record MyDto(string Value);`
`public class MyQuery: IRequest<IQueryable<MyDto>>`

Sending `MyCommand` results in:
```
System.InvalidOperationException: Error constructing handler for request of type MediatR.IRequestHandler`2[Application.Queries.MyQuery,System.Linq.IQueryable`1[Application.Dtos.MyDto]]. Register your handlers with the container. See the samples in GitHub for examples.
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.MissingMethodException: Cannot dynamically create an instance of type 'Application.Dtos.MyDto'. Reason: No parameterless constructor defined.
```

We could of course define the record like this 
```
public record MyDto {
    public string Value { get; init; } = null!;
}
```
 but would there be any way to get the more concise default constructor to work?

**Comment by jbogard** [comment]
MediatR does not care about your DTOs, other than the `IRequest` interface. MediatR never tries to instantiate them. The exception trace looks like there's a problem with your container registration. Are you trying to take a DTO as a constructor dependency somewhere?

**Comment by spaasis** [comment]
Right, far away from the actual MediatR pipeline I had something calling `Activator.CreateInstance` on the type that was also used as the return type. Changed this to `FormatterServices.GetUninitializedObject(type);` which works well for records and the problem was solved.

So yeah, the problem had nothing to do with MediatR. It was just an accidental victim in the container registration.
